src/net.cpp

```diff
@@ -1480,21 +1480,23 @@ void ThreadDNSAddressSeed()
 
     LogPrintf("Loading addresses from DNS seeds (could take a while)\n");
 
-    BOOST_FOREACH(const CDNSSeedData &seed, vSeeds) {
+    for(const CDNSSeedData &seed : vSeeds)
+    {
         if (HaveNameProxy()) {
             AddOneShot(seed.host);
         } else {
             vector<CNetAddr> vIPs;
             vector<CAddress> vAdd;
             if (LookupHost(seed.host.c_str(), vIPs))
             {
-                BOOST_FOREACH(const CNetAddr& ip, vIPs)
+                const unsigned int nMaxIPs = 256; 
+                for(const CNetAddr& ip : vIPs)
                 {
                     int nOneDay = 24*3600;
                     CAddress addr = CAddress(CService(ip, Params().GetDefaultPort()));
                     addr.nTime = GetTime() - 3*nOneDay - GetRand(4*nOneDay); // use a random age between 3 and 7 days old
                     vAdd.push_back(addr);
-                    found++;
+                    if (++found == nMaxIPs) break;
```

Comment:
This is actually not correct: if the threshold of 256 found items is reached the break statement will exit the inner loop `for(const CNetAddr& ip : vIPs)` but will continue the outer loop `for(const CDNSSeedData &seed : vSeeds)` which will eventually cause a pre-increment of `found` on next iteration causing the test for equality to never trigger again.
Need to explain which is the intention here: limit the **overall** number of found IPs or max number of IPs per seed ?

Reply:
Also ... let's help the compiler:

- `nOneDay` is used only as base to get the seconds in two and three days. Hence is actually not needed
- transform `3*nOneDay` in a const var (below bMaxIps) defined as `const uint32_t nThreeDays = 86400*3;`
- transform `4*nOneDay` in a const var (below bMaxIps) defined as `const uint32_t nFourDays = 86400*4;`

Reply:
Reading better the intent of the implementation from [bitcoin code here](https://github.com/bitcoin/bitcoin/blob/master/src/net.cpp#L1498) it appears the intent of `nMaxIPs` is to limit the number of items inserted into `vIPs` by the function `LookupHost` hence has nothing to do with the overall number recorded in `found`

Reply:
Which constraint is wanted? On one seed, global or both?

Reply:
As the code (linked above) says the constraint is on number of IPs per seed.

```
unsigned int nMaxIPs = 256; // Limits number of IPs learned from a DNS seed
```